### PR TITLE
Adds timedout check to Chatbot find/create user

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?
Begins work on #824 and #849 by splitting out checks for finding/creating a User for a given phone number, and adding `req.timedout` checks per #852.

#### How should this be reviewed?
* Use existing user, create new user
* Test with local connectivity/timeout settings to verify request will timeout:
```
9:50:35 AM web.1 |  debug: User.lookup(mobile, [removed])
9:50:47 AM web.1 |  debug: northstar.Users.lookup success
9:50:47 AM web.1 |  error: Request timed out after 2 seconds.
```
#### Any background context you want to provide?
Next up: splitting out loading Campaign and Signup into smaller functions. 

#### Relevant tickets
#824, #849, #852

#### Checklist
- [x] Tested on staging.

### CODE REVIEW

File: app/routes/chatbot.js [hunk 1]
```diff
@@ -44,58 +45,16 @@ function isCommand(incomingMessage, commandType) {
...
-    incomingMessage: scope.incoming_message,
-    keyword: scope.keyword,
-    mobileCommonsBroadcastId: scope.broadcast_id,
-    mobileCommonsMessageId: scope.body.message_id,
-    mobileCommonsProfileId: req.body.profile_id,
-  });
-
-  let configured = true;
-  // Check for required config variables.
+router.use((req, res, next) => {
```
Comment (sergii-tkachenko) [inline app/routes/chatbot.js]: A question: won't this middleware be executed on every request, despite actual route?

Comment (aaronschachter) [inline app/routes/chatbot.js]: I commented out my `MOBILECOMMONS_OIP_CHATBOT` variable and confirmed this function doesn't execute when posting to the Signups endpoint.

Comment (aaronschachter) [inline app/routes/chatbot.js]: From https://expressjs.com/en/api.html#router:
> The top-level express object has a Router() method that creates a new router object.

We declare `express.Router()` in each route, which seems to be reason why they are separate. 

https://github.com/DoSomething/gambit/blob/3.10.0/app/routes/chatbot.js#L7
https://github.com/DoSomething/gambit/blob/3.10.0/app/routes/signups.js#L4

Comment (sergii-tkachenko) [inline app/routes/chatbot.js]: Perfect.

**Review by sergiitk — APPROVED** [review]
👍

**Pull request merged**